### DISCUSSION TITLE
Problem Importing Simple Function From "LibraryFolder" To "Main"

### DISCUSSION THREAD

**Discussion by Eloufou** [micropython/micropython#14157] [RP2040 / Pico]
***Just realized its in the wrong section. im using RP2040 tho***

I want to create a simple code to test if I am able to import a library with some function from the "LibraryFolder.py" to the "main.py" but it does not work. I checked in several GitHub Forum but can't find anything...

here is the "main.py"
```
from LibraryFolder import blink_led

# Call the function to start blinking the LED
blink_led()
```

and here is the "LibraryFolder.py"

```
from machine import Pin
from utime import sleep

def blink_led():
    pin = Pin("LED", Pin.OUT)
    print("LED starts flashing...")
    while True:
        try:
            pin.toggle()
            sleep(1) # sleep 1sec
        except KeyboardInterrupt:
            break
    pin.off()
    print("Finished.")
```

and here is the Error

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: no module named 'LibraryFolder'
```

Im coding in VSCode V1.87.2 with Python V3.12.2

Thanks for your help!

**Comment by ned-pcs** [comment]
`os.listdir()` will list your files. Are both of your files present where you expect them to be?

`sys.path` will show what the import path is. Is the directory where those files are on the path?

**Reply by Eloufou** [reply]
I’ll try to see if my file are in the right place!

I’ll come back with some answers.

thanks alot

**Reply by Eloufou** [reply]
I used another method to check for the files official location and both the "LibraryFolder.py" and the "main.py" ("test.py") are in the same folder (C:\Users\Eloi\OneDrive\Documents\PlatformIO\Projects\Arduino\src\Pico). 

Any other ideas?
![Capture d’écran 2024-03-22 162713](https://github.com/micropython/micropython/assets/97297284/965089ae-3d08-406c-b61b-2b2a0c750527)

**Reply by robert-hh** [reply]
You have to copy the files to the RP2 board. Code running on the board cannot find files on your PC.

**Reply by robert-hh** [reply]
As an alternative, use mpremote with it's `mount` feature, which mounts directories on the PC as local directories on the board.

**Reply by Eloufou** [reply]
ill try this

**Reply by Eloufou** [reply]
I need to put the .U2F file in the RP2040? If so, do I need to take the one for the 2 codes and where do I find them?

**Reply by robert-hh** [reply]
You have to copy the Python files to the board. The .uf2 file is only uploaded by the bootloader during firmware update. With the bootlaoder you cannot upload Python files. 
A tool to upload python Files is for instance mpremote. If you work with Thonny (it seems to be the case), then you may as well use Thonny for that task. But I highly reccomend mpremote, since that is the official tool for MicroPython.

**Reply by robert-hh** [reply]
For mpremote, see https://docs.micropython.org/en/latest/reference/mpremote.html
